src/Libraries/Microsoft.Extensions.Http.AutoClient/StaticHeaderAttribute.cs

```diff
@@ -33,8 +35,8 @@ public sealed class StaticHeaderAttribute : Attribute
     /// <summary>
     /// Initializes a new instance of the <see cref="StaticHeaderAttribute"/> class.
     /// </summary>
-    /// <param name="header">The name of the header.</param>
-    /// <param name="value">The value of the header.</param>
+    /// <param name="header">The name of the header. Cannot be empty or null.</param>
+    /// <param name="value">The value of the header. If null, the header will not be sent.</param>
     public StaticHeaderAttribute(string header, string value)
```

Comment:
Should this be `string? value`, given the comment you just added?

Reply:
Or maybe `string? value = null`

Reply:
Sounds good. 

I went for this solution on the "Value" specifically because it's what happens with dynamic ones as well. If you pass null value to the method the header/query will not be sent.

But for static we can definitely validate and just not allow it. I'll do that :)

Reply:
Fixed!